src/main/java/model/data/process/functions/RowFunction.java

```diff
@@ -0,0 +1,83 @@
+package model.data.process.functions;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import model.data.DataRow;
+import model.data.DataTable;
+import model.data.describer.DataDescriber;
+import model.data.value.FloatValue;
+import model.data.value.IntValue;
+import model.data.value.NumberValue;
+
+/**
+ * This class will provide a framework for functions resulting multiple rows of data
+ * @author louisgosschalk
+ *13-05-2015
+ */
+public class RowFunction extends Function{
+	protected boolean minimum;
+	protected boolean maximum;
+	private DataTable table;
+	private DataDescriber<NumberValue> argument;
+	private List<DataRow> rowlist;
+	private DataRow row;
+	
+	public RowFunction(DataTable model, DataDescriber<NumberValue> argument) {
+		super(model, argument);
+		this.table = model;
+		this.argument = argument;
+		this.minimum = false;
+		this.maximum = false;
+		this.rowlist = new ArrayList<DataRow>();
+		this.row = null;
+	}
+	/**
+	 * this function checks restrictions for the calculation and calls its execution
+	 * @return List<DataRow> the rows containing the minimum
+	 */
+	public List<DataRow> calculate() {
+		if(table.getRowCount() == 0)
+			return rowlist; 
+			
+		if(table.getRowCount() == 1) {
```

Comment:
niet fout, maar wel overbodig. Stel je haalt dit weg, dan zet regel 47 het resultaat op regel0. De conditie voor de for-loop is meteen false (mits de bug wordt opgelost), dus dan wordt alsnog row0 als resultaat terug gegeven.